src/DynamoCore/Graph/Workspaces/WorkspaceModel.cs

```diff
@@ -991,6 +991,20 @@ public virtual void Save(string filePath, bool isBackup = false, EngineControlle
             try
             {
                 // Stage 1: Serialize the workspace.
+                string fileName = string.Empty;
+                try
+                {
+                    fileName = Path.GetFileName(filePath);
+                    string extension = Path.GetExtension(filePath);
+                    if (extension == ".dyn" || extension == ".dyf")
```

Comment:
is there any other extension possible?

Reply:
This is tricky here in that a file can be saved with any extension, but a file can also contain a "." character which can also be treated as an extension. I forget where but there is already a precedent for just dealing with "*.dyn" and "*.dyf" extensions specifically. I believe that doing this check here in this way is probably the best way to go (at least for now).